### PR TITLE
Remove encode_structured_data

### PR DESCRIPTION
**Pull request by kclowes**
### What was wrong?
`encode_structured_data` has been deprecated in favor of `encode_typed_data`.

### How was it fixed?
Removed all references to `encode_structured_data`

### Todo:

- [x] Clean up commit history
- [x] Add or update documentation related to these changes
- [x] Add entry to the [release notes](https://github.com/ethereum/eth-account/blob/main/newsfragments/README.md)

#### Cute Animal Picture

Put a link to a cute animal picture inside the parenthesis-->

### CODE REVIEW

**Review by reedsa — APPROVED** [review]
Lgtm!

**Comment by pacrob** [comment]
I'm pretty sure the whole `eth_account/_utils/structured_data` folder can go along with this - `encode_typed_data` has it's own folder in `_utils`.

**Comment by kclowes** [comment]
> I'm pretty sure the whole eth_account/_utils/structured_data folder can go along with this - encode_typed_data has it's own folder in _utils. And tests/core/test_structured_data_signing.py.

Ah, yep. Nice catch!

File: pyproject.toml [hunk 1]
```diff
@@ -66,6 +66,7 @@ addopts = "-v --showlocals --durations 10"
 xfail_strict = true
 log_format = "%(levelname)8s  %(asctime)s  %(filename)20s  %(message)s"
 log_date_format = "%m-%d %H:%M:%S"
+markers = "compatibility"
```
Comment (pacrob) [inline pyproject.toml:69]: ~~what's this change related to?~~ Nevermind, found it.

**Review by pacrob — APPROVED** [review]
lgtm!

**Pull request merged**